### ISSUE TITLE
Streamline the installation process 

### ISSUE THREAD

**Issue by tgodzik** [coursier/coursier#2647]
Similar to how we do it in ScalaCLI it should be possible to install coursier via different package managers:

Linux:
- [ ] Apt
- [ ] Deb
- [ ] Yum
- [ ] Rpm
- [ ] Alpine
- [ ] Nix
- [ ] SDKMAN

Windows:
- [ ]  standard installer
- [ ]  SDKMAN
- [ ] Chocolatey
- [ ] Scoop

MacOS:
- [ ] standard installer
- [ ] Brew
- [ ] Nix
- [ ] SDKMAN

If some of them are already available do mark it as such.

@lwronski might be able to help out here.

**Comment by mkurz** [comment]
+1 for SDKMAN

**Comment by doehyunbaek** [comment]
Is single script installer like [rustup](https://rustup.rs/), [pyenv](https://github.com/pyenv/pyenv-installer) a bad idea? As a person who both uses mac and linux for development, they are my favorite kinds. Just run the same scripts for both mac and linux.

**Comment by mkurz** [comment]
Just to let you know, for sdkman you have to use an API to tell them about new releases: https://sdkman.io/vendors
You can take a look how scala-cli handles that, they automatically ping the api via the GitHub actions workflow that handles new releases.

**Comment by julienrf** [comment]
> Is single script installer like [rustup](https://rustup.rs/), [pyenv](https://github.com/pyenv/pyenv-installer) a bad idea? As a person who both uses mac and linux for development, they are my favorite kinds. Just run the same scripts for both mac and linux.

I agree that this is a good approach. In addition to providing packages for the systems listed above, the “manual” installation process could be further simplified. Currently, there are many branches depending on your OS and architecture. Instead, macOS and Linux users should just run a `curl https://scala-lang.org/setup.sh | sh` that would automatically download the right `cs` binary, run `cs setup`, and `rm cs`.

**Comment by philwalk** [comment]
> macOS and Linux users should just run a `curl https://scala-lang.org/setup.sh | sh` that would automatically download the right `cs` binary, run `cs setup`, and `rm cs`.

Ideally we can also support users running in `Windows` from a `bash` shell with `OSTYPE` defined (such as [Git for Windows](https://gitforwindows.org/)).

**Comment by serpro69** [comment]
Hi,
Is there any progress on being able to install coursier via sdkman? There's a candidate available, but no actual releases (versions) yet.
Thanks

**Comment by tgodzik** [comment]
No progress currently as far as I know. We don't have a person currently to work on this.

**Comment by eed3si9n** [comment]
See https://github.com/coursier/coursier/issues/2953

I wanted `cs setup` to work, but it seems to install some unauthentic program as `sbt` on user's machine. I kindly request sbt to be taken out of the list of program Coursier attempts to install if it's unable or unwilling to install bit-by-bit identical binaries as sbt installer would set up on macOS, Linux, or Windows.

**Comment by tgodzik** [comment]
There might be something wrong here https://github.com/coursier/apps/blob/main/apps/resources/sbt.json#L16C20-L16C95

That should be the correct thing, but it's not pointing to the architecture. So maybe that's the thing to fix?

**Comment by tgodzik** [comment]
Och wait, why the type is `  "launcherType": "graalvm-native-image",` that is not native image. Something is weird here :thinking: There is an option that coursier will build native image from jars, but there aren't any jar deps here in the configuration.

**Comment by eed3si9n** [comment]
`https://github.com/sbt/sbt/releases/download/v${version}/sbt-${version}.zip` looks good to me. It's the other parts like:
```
  "dependencies": [
    "org.scala-sbt:sbt-launch:latest.stable",
    "io.get-coursier.sbt:sbt-runner:latest.stable"
  ],
```

which I don't know what it's doing, including the GraalVM thing.

**Comment by tgodzik** [comment]
ach, I think it's building native image on demand based on the launcher jar and sbt-runner one. I think that's always a problematic option. And this is what is being actually run https://github.com/coursier/sbt-runner

I am not sure what's the history here. Maybe we should just make sbt a normal jar based app.

**Comment by eed3si9n** [comment]
sbt I think should be considered a shell script that happens to call `java` or `sbtn` (GraalVM native image of sbt client) inside it.

**Comment by adpi2** [comment]
> which I don't know what it's doing, including the GraalVM thing.

It's all about the forward compatibility of older Coursier versions. So  an old Coursier would download the `coursier/sbt-runner\ and build an native image. But a newer Coursier downloads and installs the official zip. But even in that case I think creates some sort of indirection files that points to the shell script.

**Comment by shian15810** [comment]
Just to add on to my use case.

If Coursier is on SDKMAN, it would be so easy to make use of SDKMAN-installed Coursier, to create a VSCode devcontainer for a robust Scala development environment in container.

I guess others like me who develop using devcontainer, who use Dockerfile with SDKMAN already installed, plus CI/CD and DevOps in general would greatly benefit.
